File: 1.1-array/array.c
#include <stdio.h>

int main()
{
    int arr[] = {1, 2, 3, 4};
    printf("%d\n", &arr[0]); // 6422288
    printf("%d\n", &arr[1]); // 6422292
    printf("%d\n", &arr[2]); // 6422296
    printf("%d\n", &arr[3]); // 6422300
    return 0;
}

File: 1.1-array/array.js
const arr = [5, 4, 3, 2, 1]

Array.prototype.inReversed = function () {
    const reverse = (start, end) => {
        if (start > end) return;
        const tmp = this[start];
        this[start] = this[end];
        this[end] = tmp;
        reverse(start + 1, end - 1);
    };
    reverse(0, this.length - 1);
};
arr.inReversed()
console.log(arr);